File: backend/app/schemas/cost_estimation.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Dict, Literal
from uuid import UUID
from datetime import datetime


class DesignParamItem(BaseModel):
    """单个设计参数"""
    value: float
    unit: str
    description: Optional[str] = None


class CostEstimateItem(BaseModel):
    """分项估算明细"""
    category: str  # "土方工程"
    item: str  # "土方开挖"
    quantity: float
    unit: str  # "m³"
    unit_price: float
    subtotal: float


class CostEstimateSummary(BaseModel):
    """分类汇总"""
    category: str
    total_quantity: float
    total_amount: float


class CostEstimateCreateRequest(BaseModel):
    project_type: Literal["堤防", "河道整治"] = "堤防"
    design_params: Dict[str, float]


class CostEstimateResponse(BaseModel):
    """估算响应"""
    id: UUID
    project_id: UUID
    project_type: str
    version: int
    status: str
    design_params: Dict[str, float]
    summary: List[CostEstimateSummary]
    details: List[CostEstimateItem]
    total_cost: float
    cost_per_km: Optional[float] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CostEstimateListResponse(BaseModel):
    """估算列表响应"""
    estimates: List[CostEstimateResponse]
    total: int

File: frontend/src/api/cost_estimation.ts
import client from './client'

export interface CostEstimateItem {
  category: string
  item: string
  quantity: number
  unit: string
  unit_price: number
  subtotal: number
}

export interface CostEstimateSummary {
  category: string
  total_quantity: number
  total_amount: number
}

export interface CostEstimate {
  id: string
  project_id: string
  project_type: string
  version: number
  status: string
  design_params: Record<string, number>
  summary: CostEstimateSummary[]
  details: CostEstimateItem[]
  total_cost: number
  cost_per_km: number | null
  created_at: string
}

export interface CostEstimateListResponse {
  estimates: CostEstimate[]
  total: number
}

export interface CostEstimateCreateRequest {
  project_type: '堤防' | '河道整治'
  design_params: Record<string, number>
}

export const costEstimationApi = {
  create: (projectId: string, data: CostEstimateCreateRequest) =>
    client.post<CostEstimate>(`/projects/${projectId}/cost-estimates`, data),

  list: (projectId: string) =>
    client.get<CostEstimateListResponse>(`/projects/${projectId}/cost-estimates`),

  get: (projectId: string, estimateId: string) =>
    client.get<CostEstimate>(`/projects/${projectId}/cost-estimates/${estimateId}`),

  recalculate: (projectId: string, estimateId: string) =>
    client.post<CostEstimate>(`/projects/${projectId}/cost-estimates/${estimateId}/recalculate`),
}
